src/kudu/consensus/log_index.cc

```diff
@@ -128,10 +136,23 @@ Status LogIndex::IndexChunk::Open() {
   RETURN_NOT_OK(CheckError(fd_, "open"));
 
   int err;
-  RETRY_ON_EINTR(err, ftruncate(fd_, kChunkFileSize));
+  RETRY_ON_EINTR(err, ftruncate(fd_, size_));
   RETURN_NOT_OK(CheckError(fd_, "truncate"));
 
-  mapping_ = static_cast<uint8_t*>(mmap(nullptr, kChunkFileSize, PROT_READ | PROT_WRITE,
+  return Status::OK();
```

Comment:
to keep the interface compatible, should Open make a tail call to Mmap() ?

Reply:
I thought about it too, but went against it. The main reason being that it is called from `OpenAllChunksOnStartup()` and I do not want to add an argument to `Open()` indicating that the file should be mmapped or not. `OpenAllChunksOnStartup() ` should not mmap index chunks on opening all files.